Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -2218,7 +2194,30 @@ private void UpdateResourceAvailability(PerformanceCollector.MachinePerfInfo per
                     }
                 }
 
+#if PLATFORM_OSX
+                Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
+                var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
+                var startCancellingPips = false;
+
+                if (result)
+                {
+                    // If the memory pressure level is above the configured level we start canceling pips to avoid Jetsam to kill our process
+                    startCancellingPips = pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel;
+                }
+                else
+                {
+                    Logger.Log.UnableToGetMemoryPressureLevel(
+                            m_executePhaseLoggingContext,
+                            availableRam: perfInfo.AvailableRamMb.Value,
+                            minimumAvailableRam: m_configuration.Schedule.MinimumTotalAvailableRamMb,
+                            ramUtilization: perfInfo.RamUsagePercentage.Value,
+                            maximumRamUtilization: m_configuration.Schedule.MaximumRamUtilizationPercentage);
+                }
+
+                if (!m_scheduleConfiguration.DisableProcessRetryOnResourceExhaustion && startCancellingPips)
+#else
```

Comment:
It looks kind of ugly to have these dangling `if`s inside of if/def sections.

Instead, I'd write it like
```csharp
bool startCancellingPips = !resourceAvailable;
#if PLATFORM_OSX
    // special OSX logic for computing whether to start cancelling pips
    startCancellingPips = ...;
#endif

if (!m_scheduleConfiguration.DisableProcessRetryOnResourceExhaustion && startCancellingPips)
{
...
}
```